### ISSUE TITLE
puppeteer-extra-plugin-stealth www.hyatt.com search bot detection

### ISSUE THREAD

**Issue by superwunc** [berstend/puppeteer-extra#29]
If I normally browse the following link, it doesn't throw any errors, however when I use puppeteer, it throws following error.
https://www.hyatt.com/en-US/search/New%20York?checkinDate=2019-04-05&checkoutDate=2019-04-06&rooms=1&adults=1&kids=0&rate=Standard
<img width="688" alt="1554175658(1)" src="https://user-images.githubusercontent.com/1547458/55374249-5ee46000-553a-11e9-99a7-2c9feb0e808b.png">
<img width="781" alt="1554175620(1)" src="https://user-images.githubusercontent.com/1547458/55374269-6b68b880-553a-11e9-8db5-f8a973150fc1.png">

first, is the manually , second is the puppeteer and puppeteer-extra-plugin-stealth,
The system is Windows 10
 "puppeteer": "^1.12.2",
  "puppeteer-extra": "^2.1.3",
  "puppeteer-extra-plugin-stealth": "^2.2.2",

The code is as follows
`const puppeteer = require("puppeteer-extra")

const pluginStealth = require("puppeteer-extra-plugin-stealth")

puppeteer.use(pluginStealth())
puppeteer.launch({ headless: false }).then(async browser => {
  const page = await browser.newPage();
  await page.goto('https://www.hyatt.com/en-US/search/New%20York?checkinDate=2019-04-05&checkoutDate=2019-04-06&rooms=1&adults=1&kids=0&rate=Standard');
  await page.waitFor(5000)
  await page.screenshot({ path: "testresult2.png", fullPage: true })
  //await browser.close()
 // await browser.close();
});`

**Comment by superwunc** [comment]
e: {"stack":"TypeError: getParameter is not a function\n    at WebGLRenderingContext.getParameter (<anonymous>:14:18)\n    at e.getWebglFp (https://www.hyatt.com/149e9513-01fa-4fb0-aad4-566afd725d1b/2d206a39-8ed7-437e-a3be-862e0f06eea3/f.js:1:19472)\n    at e.webglKey (https://www.hyatt.com/149e9513-01fa-4fb0-aad4-566afd725d1b/2d206a39-8ed7-437e-a3be-862e0f06eea3/f.js:1:4766)\n    at e.get (https://www.hyatt.com/149e9513-01fa-4fb0-aad4-566afd725d1b/2d206a39-8ed7-437e-a3be-862e0f06eea3/f.js:1:1436)\n    at _0x211bc5 (https://www.hyatt.com/149e9513-01fa-4fb0-aad4-566afd725d1b/2d206a39-8ed7-437e-a3be-862e0f06eea3/fingerprint/script/kpf.js?url=/149e9513-01fa-4fb0-aad4-566afd725d1b/2d206a39-8ed7-437e-a3be-862e0f06eea3/fingerprint&token=a2e05d5f-b10d-265d-6822-076cc0e423b3:1:23611)\n    at https://www.hyatt.com/149e9513-01fa-4fb0-aad4-566afd725d1b/2d206a39-8ed7-437e-a3be-862e0f06eea3/fingerprint/script/kpf.js?url=/149e9513-01fa-4fb0-aad4-566afd725d1b/2d206a39-8ed7-437e-a3be-862e0f06eea3/fingerprint&token=a2e05d5f-b10d-265d-6822-076cc0e423b3:1:24499\n    at https://www.hyatt.com/149e9513-01fa-4fb0-aad4-566afd725d1b/2d206a39-8ed7-437e-a3be-862e0f06eea3/fingerprint/script/kpf.js?url=/149e9513-01fa-4fb0-aad4-566afd725d1b/2d206a39-8ed7-437e-a3be-862e0f06eea3/fingerprint&token=a2e05d5f-b10d-265d-6822-076cc0e423b3:1:26092","message":"getParameter is not a function","name":"TypeError"}

**Comment by superwunc** [comment]
I find a solution 
--disable-webgl will block webgl fingerprint

const puppeteer = require("puppeteer-extra")
const pluginStealth = require("puppeteer-extra-plugin-stealth")
puppeteer.use(pluginStealth())
const args = [
    '--disable-webgl'
];
puppeteer.launch({args:args,  headless: false }).then(async browser => {
  const page = await browser.newPage();
  await page.goto('https://www.hyatt.com/en-US/search/New%20York?checkinDate=2019-04-05&checkoutDate=2019-04-06&rooms=1&adults=1&kids=0&rate=Standard');
  await page.waitFor(5000)
  await page.screenshot({ path: "testresult2.png", fullPage: true })
  //await browser.close()
});

**Comment by antonioiliev** [comment]
Can someone explain why this solution works? 

I was being blocked by Ebay with the same error, however adding this argument `const args = [
'--disable-webgl'
];` actually fixed it... The worst part is - I do not understand why...

**Comment by superwunc** [comment]
https://fingerprintjs.com  https://fingerprintjs.com/demo  can check client is bot ,  --disable-webgl can block webgl feature fingerprint check

**Comment by berstend** [comment]
Please checkout `v2.4.0` of the stealth plugin ([Changelog](https://github.com/berstend/puppeteer-extra/tree/master/packages/puppeteer-extra-plugin-stealth#v240)) which contains a boat load of other fixes as well. :)

Closing this for now, please open a new issue if you experience any issues with the new release.
